src/Core/Jobs/BaseJobsHostedService.cs

```diff
@@ -18,25 +18,45 @@ public abstract class BaseJobsHostedService : IHostedService, IDisposable
         protected readonly ILogger _logger;
 
         private IScheduler _scheduler;
+        protected GlobalSettings _globalSettings;
 
         public BaseJobsHostedService(
+            GlobalSettings globalSettings,
             IServiceProvider serviceProvider,
             ILogger logger,
             ILogger<JobListener> listenerLogger)
         {
             _serviceProvider = serviceProvider;
             _logger = logger;
             _listenerLogger = listenerLogger;
+            _globalSettings = globalSettings;
         }
 
         public IEnumerable<Tuple<Type, ITrigger>> Jobs { get; protected set; }
 
         public virtual async Task StartAsync(CancellationToken cancellationToken)
         {
-            var factory = new StdSchedulerFactory(new NameValueCollection
+            var props = new NameValueCollection
             {
-                { "quartz.serializer.type", "binary" }
-            });
+                {"quartz.serializer.type", "binary"},
+            };
+
+            if (!_globalSettings.SelfHosted && !string.IsNullOrEmpty(_globalSettings.SqlServer.JobSchedulerConnectionString))
+            {
+                // Ensure each project has a unique instanceName
+                props.Add("quartz.scheduler.instanceName", GetType().FullName);
+                props.Add("quartz.scheduler.instanceId", "AUTO");
+                props.Add("quartz.jobStore.type", "Quartz.Impl.AdoJobStore.JobStoreTX");
+                props.Add("quartz.jobStore.driverDelegateType", "Quartz.Impl.AdoJobStore.SqlServerDelegate");
+                props.Add("quartz.jobStore.useProperties", "true");
+                props.Add("quartz.jobStore.dataSource", "default");
+                props.Add("quartz.jobStore.tablePrefix", "QRTZ_");
+                props.Add("quartz.jobStore.clustered", "true");
+                props.Add("quartz.dataSource.default.provider", "SqlServer");
+                props.Add("quartz.dataSource.default.connectionString", _globalSettings.SqlServer.JobSchedulerConnectionString);
```

Comment:
string => string KVP? They don't have an object with properties that you configure for this stuff?  Looks pretty gross.

Reply:
I don't believe we have any other methods than `Add` available for this.

Would something like this look better:
```csharp
var props = new NameValueCollection();

if (!string.IsNullOrEmpty(_globalSettings.SqlServer.JobSchedulerConnectionString))
{
    props = new NameValueCollection
    {
        // Ensure each project has a unique instanceName
        {"quartz.scheduler.instanceName", GetType().FullName},
        {"quartz.scheduler.instanceId", "AUTO"},
        {"quartz.jobStore.type", "Quartz.Impl.AdoJobStore.JobStoreTX"},
        {"quartz.jobStore.driverDelegateType", "Quartz.Impl.AdoJobStore.SqlServerDelegate"},
        {"quartz.jobStore.useProperties", "true"},
        {"quartz.jobStore.dataSource", "default"},
        {"quartz.jobStore.tablePrefix", "QRTZ_"},
        {"quartz.jobStore.clustered", "true"},
        {"quartz.dataSource.default.provider", "SqlServer"},
        {
            "quartz.dataSource.default.connectionString",
            _globalSettings.SqlServer.JobSchedulerConnectionString
        },
    };
}

props.Add("quartz.serializer.type", "binary");
```

Reply:
Yes, that would work